packages/arb-validator-core/gobridge/arbRollup.go

```diff
@@ -630,38 +691,47 @@ func (vm *arbRollup) StartChallenge(
 	challenger.inChallenge = true
 
 	// generate contractAddress
-	newAddr := eth.getNextAddress()
-	eth.challenges[newAddr] = &challengeData{deadline: common.TimeTicks{disputableDeadline}, challengerDataHash: challengerDataHash}
+	challengeAddress, _ := eth.challengeFactoryContract.CreateChallenge(
+		ctx,
+		asserterAddress,
+		challengerAddress,
+		challengerPeriodTicks,
+		challengerDataHash,
+		new(big.Int).SetUint64(uint64(challengerPosition)),
+	)
+	//newAddr := eth.getNextAddress()
+	//eth.challenges[newAddr] = &challengeData{deadline: common.TimeTicks{disputableDeadline}, challengerDataHash: challengerDataHash}
 	// initialize bisection
 	//save data
 	// deadline = current + challenge period
-	eth.challenges[newAddr].deadline = common.TicksFromBlockNum(eth.LastMinedBlock.Height).Add(challengerPeriodTicks)
+	//eth.challenges[newAddr].deadline = common.TicksFromBlockNum(eth.LastMinedBlock.Height).Add(challengerPeriodTicks)
+	eth.challenges[challengeAddress].challengeData.deadline = common.TicksFromBlockNum(eth.getCurrentBlock().Height).Add(challengerPeriodTicks)
 	// emit InitiatedChallenge
```

Comment:
This stuff seems unnecessary. If I remember correctly the deadline is set in the challenge constructor.

Reply:
fixed